File: app/src/main/java/com/beatrice/bookfinder/MainFragment.java
package com.beatrice.bookfinder;


import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.NavDestination;
import androidx.navigation.NavHost;
import androidx.navigation.Navigation;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.bottomnavigation.BottomNavigationView;


/**
 * A simple {@link Fragment} subclass.
 */
public class MainFragment extends Fragment {


    public MainFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_main, container, false);
        NavHostFragment mNavHostFragment = (NavHostFragment) getChildFragmentManager().findFragmentById(R.id.fragment_container_main);
        NavController mNavController = NavHostFragment.findNavController(mNavHostFragment);

        //set up BottomNavigation using setUpWithNavController of navigation components
        BottomNavigationView mBottomNav = rootView.findViewById(R.id.bottomNavigationView);
        NavigationUI.setupWithNavController(mBottomNav, mNavController);

        //set up Toolbar using setUpWithNavController of navigation components
        final Toolbar mToolBar = rootView.findViewById(R.id.toolbar);
        ((AppCompatActivity)getActivity()).setSupportActionBar(mToolBar);
        //NavigationUI.setupWithNavController(mToolBar, mNavController);
        mNavHostFragment.getNavController().addOnDestinationChangedListener(new NavController.OnDestinationChangedListener() {
            @Override
            public void onDestinationChanged(@NonNull NavController controller, @NonNull NavDestination destination, @Nullable Bundle arguments) {
                mToolBar.setTitle(destination.getLabel());
            }
        });

        return  rootView;
    }

}

File: app/src/main/java/com/beatrice/bookfinder/MainFragment.kt
package com.beatrice.bookfinder


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.widget.Toolbar
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupWithNavController
import com.google.android.material.bottomnavigation.BottomNavigationView

/**
 * A simple [Fragment] subclass.
 */
class MainFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val rootView =  inflater.inflate(R.layout.fragment_main, container, false)

        val msgTextview = rootView.findViewById<TextView>(R.id.msg_txtview)

        //passing text using bundles
        //val name = arguments?.getString("name")

        //passing data using safe args
        //use fromBundles() method to retrieve the arguments
        val name = MainFragmentArgs.fromBundle(arguments!!).name
        msgTextview.text ="welcome " +name


        //navHostFragment
        val navHostFragment = childFragmentManager.findFragmentById(R.id.fragment_container_main) as NavHostFragment
        //set up bottomNavigationView
        rootView.findViewById<BottomNavigationView>(R.id.bottomNavigationView).setupWithNavController(
                navController = navHostFragment.navController
        )

        //set up toolbar
//        rootView.findViewById<Toolbar>(R.id.toolbar).setupWithNavController(
//                navController = navHostFragment.navController
//        )

//        //setting up toolbar using custom navigatedListener
        val toolbar = rootView.findViewById<Toolbar>(R.id.toolbar)
        navHostFragment.navController.addOnDestinationChangedListener { _, destination, _ ->
            toolbar.title = destination.label
        }


        return  rootView
    }


}
